src/ext/java/org/opentripplanner/ext/fares/model/FareProductInstance.java

```diff
@@ -0,0 +1,8 @@
+package org.opentripplanner.ext.fares.model;
+
+/**
+ * An instance of a fare product that might be shared across several legs.
+ * <p>
+ * The instance id identifies the fare product across the legs.
+ */
+public record FareProductInstance(String instanceId, FareProduct product) {}
```

Comment:
OO wise, this looks a bit confusing. There are instances of  FareProduct, and then this FareProductInstance. 

I wonder what could be a better name, maybe FareProductRide or something?

Reply:
I agree that the term "instance" is overloaded in OO. 

Lets brainstorm together. I can come up with the following:

- PurchasableFareProduct
- FareProductPurchase
- DeduplicatedFareProduct
- PurchasableTicket
- TicketPurchase

The question is if we should still call it "instanceId" in the API schema or switch to something like "deduplicationId".

Reply:
- `FareProductUse`
- `FareProductUse.id`

Reply:
I also think _Offer_ work well.

Reply:
I believe that "use" better reflects the concept to be honest.

Reply:
I've renamed "instance" to "use" in 4fe75539e3eea7783a3a9231073c3fb05d5ffcd3